network/endpoint_windows.go

```diff
@@ -212,21 +216,27 @@ func (nw *network) configureHcnEndpoint(epInfo *EndpointInfo) (*hcn.HostComputeE
 		Name:               infraEpName,
 		HostComputeNetwork: nw.HnsId,
 		Dns: hcn.Dns{
-			Search:     strings.Split(epInfo.DNS.Suffix, ","),
-			ServerList: epInfo.DNS.Servers,
-			Options:    epInfo.DNS.Options,
+			Search:     strings.Split(epInfo.EndpointDNS.Suffix, ","),
+			ServerList: epInfo.EndpointDNS.Servers,
+			Options:    epInfo.EndpointDNS.Options,
 		},
 		SchemaVersion: hcn.SchemaVersion{
 			Major: hcnSchemaVersionMajor,
 			Minor: hcnSchemaVersionMinor,
 		},
-		MacAddress: epInfo.MacAddress.String(),
 	}
 
-	if endpointPolicies, err := policy.GetHcnEndpointPolicies(policy.EndpointPolicy, epInfo.Policies, epInfo.Data, epInfo.EnableSnatForDns, epInfo.EnableMultiTenancy, epInfo.NATInfo); err == nil {
-		for _, epPolicy := range endpointPolicies {
-			hcnEndpoint.Policies = append(hcnEndpoint.Policies, epPolicy)
-		}
+	// macAddress type for InfraNIC is like "60:45:bd:12:45:65"
+	// if NICType is delegatedVMNIC, convert the macaddress format
+	macAddress := epInfo.MacAddress.String()
+	if epInfo.NICType == cns.DelegatedVMNIC {
+		// convert the format of macAddress that HNS can accept, i.e, "60-45-bd-12-45-65" if NIC type is delegated NIC
+		macAddress = strings.Join(strings.Split(macAddress, ":"), "-")
+	}
+	hcnEndpoint.MacAddress = macAddress
```

Comment:
is macaddress needed for infra? I thought its required only for delegated nic transparent network

Reply:
macAddress is not needed for infra. It only requires for delegated NIC now